pennylane_lightning/core/src/simulators/lightning_kokkos/gates/BasicGateFunctors.hpp

```diff
@@ -1092,6 +1095,59 @@ void applyMultiRZ(Kokkos::View<Kokkos::complex<PrecisionT> *> arr_,
         });
 }
 
+template <class ExecutionSpace, class PrecisionT>
+void applyPauliRot(Kokkos::View<Kokkos::complex<PrecisionT> *> arr_,
+                   const std::size_t num_qubits,
+                   const std::vector<std::size_t> &wires, const bool inverse,
+                   const PrecisionT angle, const std::string &word) {
+    using ComplexT = Kokkos::complex<PrecisionT>;
+    constexpr auto IMAG = Pennylane::Util::IMAG<PrecisionT>();
+    PL_ABORT_IF_NOT(wires.size() == word.size(),
+                    "wires and word have incompatible dimensions.")
+    if (std::find_if_not(word.begin(), word.end(),
+                         [](const int w) { return w == 'Z'; }) == word.end()) {
+        applyMultiRZ<ExecutionSpace>(arr_, num_qubits, wires, inverse,
+                                     std::vector<PrecisionT>{angle});
+        return;
+    }
+    const PrecisionT c = std::cos(angle / 2);
+    const ComplexT s = ((inverse) ? IMAG : -IMAG) * std::sin(angle / 2);
+    const std::vector<ComplexT> sines = {s, IMAG * s, -s, -IMAG * s};
+    auto d_sines = vector2view(sines);
+    auto get_mask =
+        [num_qubits, &wires](
+            [[maybe_unused]] const std::function<bool(const int)> &condition) {
+            std::size_t mask{0U};
+            for (std::size_t iw = 0; iw < wires.size(); iw++) {
+                const auto bit = static_cast<std::size_t>(condition(iw));
+                mask |= bit << (num_qubits - 1 - wires[iw]);
+            }
+            return mask;
+        };
+    const std::size_t mask_xy =
+        get_mask([&word](const int a) { return word[a] != 'Z'; });
+    const std::size_t mask_y =
+        get_mask([&word](const int a) { return word[a] == 'Y'; });
+    const std::size_t mask_z =
+        get_mask([&word](const int a) { return word[a] == 'Z'; });
```

Comment:
We can probably reduce one `get_mask` operation 
```suggestion
    const std::size_t mask_y =
        get_mask([&word](const int a) { return word[a] == 'Y'; });
    const std::size_t mask_z =
        get_mask([&word](const int a) { return word[a] == 'Z'; });
   const std::size_t makk_xy = ~(mask_y | mask_z) | mask_y;
```
I am not 100% sure that this should work but it would be good to try :thinking:

Reply:
Doesn't work as is. Since this is not a bottleneck, let's keep the current implementation ok?